src/mame/drivers/rastersp.cpp

```diff
@@ -436,7 +614,128 @@ WRITE_LINE_MEMBER( rastersp_state::vblank_irq )
 		update_irq(IRQ_VBLANK, ASSERT_LINE);
 }
 
+WRITE_LINE_MEMBER( rastersp_state::duart_irq )
+{
+	update_irq(IRQ_UART, state);
+}
+
 
+WRITE_LINE_MEMBER( fbcrazy_state::trackball_rts )
+{
+	m_trackball_enabled = state;
+}
+
+/*
+	Mimic trackball unit. This uses a small PIC based interface board
+	which converts the trackball movement into RS232 data.
+	The data format is as follows:
+	Byte 0 : 0100abAB bits 7 & 6 = 01 signifies 1st byte
+	Byte 1 : 00CDEFGH bits 7 & 6 = 00 signifies 2 or 3rd byte (not identified)
+	Byte 2 : 00cdefgh
+
+	The trackball X movement is ABCDEFGH and Y movement abcdefgh
+	We do not have a PCB nor a dump of the PIC :(
+*/
+TIMER_CALLBACK_MEMBER(fbcrazy_state::trackball_timer)
+{
+	if (m_trackball_enabled)
+	{
+		if (m_trackball_ctr == 0)
+		{
+			uint16_t newxpos, newypos;
+
+			newxpos = ioport("TRACK_X")->read();
+			newypos = ioport("TRACK_Y")->read();
```

Comment:
Please don’t add run-time tag map lookups.  Use `required_ioport` to find these at object resolution time.